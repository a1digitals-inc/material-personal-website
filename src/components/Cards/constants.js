export const FRONTEND_TYPE = "FRONT-END";
export const BACKEND_TYPE = "BACK-END";
