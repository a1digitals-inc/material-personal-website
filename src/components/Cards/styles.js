import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles((theme) => ({
  root: {
    width: 345,
    height: 406,
  },
  media: {
    textAlign: "center",
    textTransform: "uppercase",
    height: 140,
  },
  avatar: {
    width: "60px",
    height: "60px",
    margin: `${theme.spacing(4)}px auto`,
  },
}));

export default useStyles;
