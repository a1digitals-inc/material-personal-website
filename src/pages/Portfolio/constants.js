export const ALL_FILTER = "ALL";
export const FRONTEND_FILTER = "FRONTEND";
export const BACKEND_FILTER = "BACKEND";
