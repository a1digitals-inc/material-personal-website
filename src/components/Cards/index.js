import React, { memo } from "react";
import PropTypes from "prop-types";
import Card from "@material-ui/core/Card";
import CardActionArea from "@material-ui/core/CardActionArea";
import CardMedia from "@material-ui/core/CardMedia";
import CardContent from "@material-ui/core/CardContent";
import CardActions from "@material-ui/core/CardActions";
import Button from "@material-ui/core/Button";
import Avatar from "@material-ui/core/Avatar";
import Typography from "@material-ui/core/Typography";
import Chip from "@material-ui/core/Chip";
import Grid from "@material-ui/core/Grid";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faDesktop, faCode } from "@fortawesome/free-solid-svg-icons";

import useStyles from "./styles";
import * as types from "./constants";

const Cards = ({ name, description, type, technologies, homepage, github }) => {
  const classes = useStyles();

  return (
    <Card className={classes.root}>
      <CardActionArea>
        <CardMedia className={classes.media}>
          <Avatar className={classes.avatar}>
            {type === types.FRONTEND_TYPE ? (
              <FontAwesomeIcon icon={faDesktop} />
            ) : (
              <FontAwesomeIcon icon={faCode} />
            )}
          </Avatar>
          <Typography gutterBottom variant="overline" component="h6">
            {type}
          </Typography>
        </CardMedia>
        <CardContent>
          <Typography gutterBottom variant="h6">
            {name}
          </Typography>
          <Typography variant="body2" color="textSecondary" component="p">
            {description}
          </Typography>
        </CardContent>
      </CardActionArea>
      <CardActions>
        <Grid container justify="flex-start" spacing={1}>
          {technologies.map((technologie) => (
            <Grid item key={technologie}>
              <Chip label={technologie} color="primary" />
            </Grid>
          ))}
        </Grid>
      </CardActions>
      <CardActions>
        {homepage && (
          <Button size="small" color="primary" target="_blank" href={homepage}>
            Homepage
          </Button>
        )}
        {github && (
          <Button size="small" color="primary" target="_blank" href={github}>
            Source code
          </Button>
        )}
      </CardActions>
    </Card>
  );
};

Cards.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string,
  type: PropTypes.oneOf([types.FRONTEND_TYPE, types.BACKEND_TYPE]).isRequired,
  technologies: PropTypes.arrayOf(PropTypes.string),
  homepage: PropTypes.string,
  github: PropTypes.string,
};

export default memo(Cards);
