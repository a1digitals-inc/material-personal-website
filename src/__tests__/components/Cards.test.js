import React from "react";
import { shallow } from "enzyme";

import Cards from "../../components/Cards";
import * as types from "../../components/Cards/constants";

it("renders without crashing", () => {
  let props;
  let wrapper;

  props = {
    name: "test 1",
    description: "test 1 description",
    type: types.FRONTEND_TYPE,
    technologies: ["technologie 1"],
    homepage: "homepage",
    github: "github",
  };
  wrapper = shallow(<Cards {...props} />);
  expect(wrapper).toMatchSnapshot();

  props = {
    name: "test 2",
    description: "test 2 description",
    type: types.BACKEND_TYPE,
    technologies: ["technologie 1"],
    homepage: "homepage",
    github: "github",
  };
  wrapper = shallow(<Cards {...props} />);
  expect(wrapper).toMatchSnapshot();

  props = {
    name: "test 3",
    description: "test 3 description",
    type: types.FRONTEND_TYPE,
  };
  wrapper = shallow(<Cards {...props} />);
  expect(wrapper).toMatchSnapshot();
});
