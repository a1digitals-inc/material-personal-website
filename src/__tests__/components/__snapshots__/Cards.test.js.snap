// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(CardActionArea))>
    <WithStyles(ForwardRef(CardMedia))
      className="makeStyles-media-2"
    >
      <WithStyles(ForwardRef(Avatar))
        className="makeStyles-avatar-3"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                576,
                512,
                Array [],
                "f108",
                "M528 0H48C21.5 0 0 21.5 0 48v320c0 26.5 21.5 48 48 48h192l-16 48h-72c-13.3 0-24 10.7-24 24s10.7 24 24 24h272c13.3 0 24-10.7 24-24s-10.7-24-24-24h-72l-16-48h192c26.5 0 48-21.5 48-48V48c0-26.5-21.5-48-48-48zm-16 352H64V64h448v288z",
              ],
              "iconName": "desktop",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </WithStyles(ForwardRef(Avatar))>
      <WithStyles(ForwardRef(Typography))
        component="h6"
        gutterBottom={true}
        variant="overline"
      >
        FRONT-END
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(CardMedia))>
    <WithStyles(ForwardRef(CardContent))>
      <WithStyles(ForwardRef(Typography))
        gutterBottom={true}
        variant="h6"
      >
        test 1
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        component="p"
        variant="body2"
      >
        test 1 description
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(CardActionArea))>
  <WithStyles(ForwardRef(CardActions))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      justify="flex-start"
      spacing={1}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        key="technologie 1"
      >
        <WithStyles(ForwardRef(Chip))
          label="technologie 1"
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(CardActions))>
  <WithStyles(ForwardRef(CardActions))>
    <WithStyles(ForwardRef(Button))
      href="homepage"
      size="small"
      target="_blank"
    >
      Homepage
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      href="github"
      size="small"
      target="_blank"
    >
      GitHub
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(CardActions))>
</WithStyles(ForwardRef(Card))>
`;

exports[`renders without crashing 2`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(CardActionArea))>
    <WithStyles(ForwardRef(CardMedia))
      className="makeStyles-media-2"
    >
      <WithStyles(ForwardRef(Avatar))
        className="makeStyles-avatar-3"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                640,
                512,
                Array [],
                "f121",
                "M278.9 511.5l-61-17.7c-6.4-1.8-10-8.5-8.2-14.9L346.2 8.7c1.8-6.4 8.5-10 14.9-8.2l61 17.7c6.4 1.8 10 8.5 8.2 14.9L293.8 503.3c-1.9 6.4-8.5 10.1-14.9 8.2zm-114-112.2l43.5-46.4c4.6-4.9 4.3-12.7-.8-17.2L117 256l90.6-79.7c5.1-4.5 5.5-12.3.8-17.2l-43.5-46.4c-4.5-4.8-12.1-5.1-17-.5L3.8 247.2c-5.1 4.7-5.1 12.8 0 17.5l144.1 135.1c4.9 4.6 12.5 4.4 17-.5zm327.2.6l144.1-135.1c5.1-4.7 5.1-12.8 0-17.5L492.1 112.1c-4.8-4.5-12.4-4.3-17 .5L431.6 159c-4.6 4.9-4.3 12.7.8 17.2L523 256l-90.6 79.7c-5.1 4.5-5.5 12.3-.8 17.2l43.5 46.4c4.5 4.9 12.1 5.1 17 .6z",
              ],
              "iconName": "code",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </WithStyles(ForwardRef(Avatar))>
      <WithStyles(ForwardRef(Typography))
        component="h6"
        gutterBottom={true}
        variant="overline"
      >
        BACK-END
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(CardMedia))>
    <WithStyles(ForwardRef(CardContent))>
      <WithStyles(ForwardRef(Typography))
        gutterBottom={true}
        variant="h6"
      >
        test 2
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        component="p"
        variant="body2"
      >
        test 2 description
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(CardActionArea))>
  <WithStyles(ForwardRef(CardActions))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      justify="flex-start"
      spacing={1}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        key="technologie 1"
      >
        <WithStyles(ForwardRef(Chip))
          label="technologie 1"
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(CardActions))>
  <WithStyles(ForwardRef(CardActions))>
    <WithStyles(ForwardRef(Button))
      href="homepage"
      size="small"
      target="_blank"
    >
      Homepage
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      href="github"
      size="small"
      target="_blank"
    >
      GitHub
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(CardActions))>
</WithStyles(ForwardRef(Card))>
`;

exports[`renders without crashing 3`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(CardActionArea))>
    <WithStyles(ForwardRef(CardMedia))
      className="makeStyles-media-2"
    >
      <WithStyles(ForwardRef(Avatar))
        className="makeStyles-avatar-3"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                576,
                512,
                Array [],
                "f108",
                "M528 0H48C21.5 0 0 21.5 0 48v320c0 26.5 21.5 48 48 48h192l-16 48h-72c-13.3 0-24 10.7-24 24s10.7 24 24 24h272c13.3 0 24-10.7 24-24s-10.7-24-24-24h-72l-16-48h192c26.5 0 48-21.5 48-48V48c0-26.5-21.5-48-48-48zm-16 352H64V64h448v288z",
              ],
              "iconName": "desktop",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </WithStyles(ForwardRef(Avatar))>
      <WithStyles(ForwardRef(Typography))
        component="h6"
        gutterBottom={true}
        variant="overline"
      >
        FRONT-END
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(CardMedia))>
    <WithStyles(ForwardRef(CardContent))>
      <WithStyles(ForwardRef(Typography))
        gutterBottom={true}
        variant="h6"
      >
        test 3
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        component="p"
        variant="body2"
      >
        test 3 description
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(CardActionArea))>
  <WithStyles(ForwardRef(CardActions))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      justify="flex-start"
      spacing={1}
    />
  </WithStyles(ForwardRef(CardActions))>
  <WithStyles(ForwardRef(CardActions)) />
</WithStyles(ForwardRef(Card))>
`;
